### PR TITLE
chore: Improves the Select component UI/UX - iteration 2

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
Improves the `Select` component UI/UX - iteration 2.

Storybook changes:

At Every Corner
- Adds a text explaining the objective of this panel
- Increases the selects width

Async Select
- Allows the user to simulate a slower connection
- Displays the dispatched requests
- Decreases the select width

Interactive Select
- Decreases the select width

Select component changes:
- Fixes a bug where the paginated scroll keeps requesting data if the user scrolls down and there is no more data available
- Moves `debounce` from `handleFetch` to `handleOnSearch`
- Renames the select

Improvements/tasks that will be done in the next iterations:

Async Select
- Make it more clear that the select is loading while scrolling
- Improve the error handling

Interactive Select
- Add the remaining properties
- Hide the options control
- When `allowNewOptions` is true, `showSearch` should be true
- Show radio group with no header, text header, and `ControlHeader`

Select component
- Check/Discuss whether we need the onPaste and on closeOnScrollOut props
- Remove the usage of any from the implementation
- Our interface should only expose single and multiple modes
- Paginate a few px before the bottom of the scroll for better UX
- Investigate why there are two `useEffects` with very similar dependencies and the same code
- Manual testing
- RTL tests

@geido @villebro @rusackas @junlincc @kgabryje The component continues to be isolated from the rest of the application. Only available in the Storybook. You don't need to test this PR, this will be done in the final iterations. The idea here is to share the current state of the work and give the opportunity to interact with the component and visualize potential improvements.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://user-images.githubusercontent.com/70410625/122447287-f696c300-cf79-11eb-8528-dc1a0e24f53f.mov

https://user-images.githubusercontent.com/70410625/122447300-fa2a4a00-cf79-11eb-951c-4a7892dd4fa5.mov

### TESTING INSTRUCTIONS
Use the Select storybook to explore all the available options.

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rusackas** [comment]
For reference, the prior iteration is here: https://github.com/apache/superset/pull/15121

File: superset-frontend/src/components/Select/Select.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+        }}
+      >
+        <Select ariaLabel={`gallery-${position.id}`} options={options} />
+      </div>
+    ))}
+    <p style={{ position: 'absolute', top: '40%', left: '33%', width: 500 }}>
+      The objective of this panel is to show how the Select behaves when in
+      touch with the viewport extremities. In particular, how the drop-down is
+      displayed and if the tooltips of truncated items are correctly positioned.
+    </p>
```
Comment (villebro) [inline superset-frontend/src/components/Select/Select.stories.tsx:87]: It would also be nice to have a story where the select can scroll

**Review by villebro — APPROVED** [review]
Looks really nice, I can't wait to migrate the native value filter to this and get lazy loading to complement the "search all filter options" option. One small observation: It would be nice to prefix the new option with "Create" when using `allowNewOptions`, like we have previously done:
image

File: superset-frontend/src/components/Select/Select.stories.tsx (hunk 1 above)
Comment (michael-s-molina) [reply superset-frontend/src/components/Select/Select.stories.tsx:87]: Great point. I'll do it in the next iteration.

**Comment by michael-s-molina** [comment]
@villebro About the "Create" when using `allowNewOptions`... I think @geido had a reason to not add this but I can't remember right now. @geido Do you remember?

**Comment by geido** [comment]
Hi @villebro @michael-s-molina . The reason not to have the "Create" is because that would be inconsistent with the defaut Antd behavior which just presents you with the value. Changing the default behavior comes with an implementation cost and we did not see the added benefit for the end user very clearly. As long as the user can create new options, it should not matter whether he is creating one or picking an existing, unless you have a different opinion.

**Pull request merged**